playground/docker-compose.fork.yml

```diff
@@ -0,0 +1,172 @@
+version: "3.9"
+services:
+  chain:
+    image: ghcr.io/foundry-rs/foundry:latest
+    restart: always
+    entrypoint: /usr/local/bin/anvil
+    command: --fork-url ${ETH_RPC_URL} --block-time 12
+    environment:
+      - ANVIL_IP_ADDR=0.0.0.0
+    ports:
+      - 8545:8545
+    healthcheck:
+      # The healthcheck is actually a little hack that replaces the allow-list manager
+      # with simple bytecode that always returns a solidity `true` value, alleviating
+      # the requirement to register a solver's private key for the sake of testing on
+      # a fork.
+      test: [
+        "CMD",
+        "/usr/local/bin/cast",
+        "rpc",
+        "-r",
+        "http://127.0.0.1:8545",
+        "anvil_setCode",
+        "0x2c4c28DDBdAc9C5E7055b4C863b72eA0149D8aFE",
+        "0x600160005260206000F3"
+      ]
+      interval: 1m30s
+      timeout: 30s
+      retries: 5
+      start_period: 5s
```

Comment:
Needed to add this to be able to use `http://host.docker.internal:8545` to reach my `reth` node running on localhost.
```
    extra_hosts:
      - "host.docker.internal:host-gateway"
```

Reply:
I would suggest when running a local node, such as `reth`, to bind it to `0.0.0.0`, and then add  an IP alias for your local NIC (or even just a "fake" bridge adaptor or similar), in which case then when running docker-in-docker, you will still be able to connect to this IP address (with `reth` on it), but avoid all the loopback catches when running docker from different contexts.